sirepo/auth_db.py

```diff
@@ -126,59 +141,66 @@ def init():
         # we access a single connection across threads
         connect_args={'check_same_thread': False},
     )
-    _session = sqlalchemy.orm.Session(bind=_engine)
+    sirepo.events.register(PKDict(
+        srcontext_created=_create_session,
+        srcontext_destroyed=_destroy_session,
+    ))
 
     @sqlalchemy.ext.declarative.as_declarative()
     class UserDbBase(object):
         STRING_ID = sqlalchemy.String(8)
         STRING_NAME =  sqlalchemy.String(100)
-        _session = _session
 
         def __init__(self, **kwargs):
             for k, v in kwargs.items():
                 setattr(self, k, v)
 
         def delete(self):
             with thread_lock:
-                self._session.delete(self)
-                self._session.commit()
+                self._session().delete(self)
+                self._session().commit()
 
         @classmethod
         def delete_all(cls):
             with thread_lock:
-                cls._session.query(cls).delete()
-                cls._session.commit()
+                cls._session().query(cls).delete()
+                cls._session().commit()
 
         def save(self):
```

Comment:
I think we should do away with save. I think the "add" should be implicit in the operation.

Reply:
We decided not to implement